CMakeLists.txt

```diff
@@ -45,56 +59,59 @@ option(IDPREFIX "Prefix CSG nodes with index # (debugging purposes only, will br
 option(PROFILE "Enable compiling with profiling / test coverage instrumentation" OFF)
 option(MXECROSS "Enable setup for MXE cross platform build" OFF)
 option(OFFLINE_DOCS "Download Documentation for offline usage" OFF)
+option(USE_MIMALLOC "Use mimalloc as malloc replacement." ON)
 
 if(APPLE)
   option(APPLE_UNIX "Build OpenSCAD in Unix mode in MaxOS X instead of an Apple Bundle" OFF)
 endif()
 
+#set_property(TARGET OpenSCAD PROPERTY CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
 set(SUFFIX "" CACHE STRING "Installation suffix for binary (e.g. 'nightly')")
 set(STACKSIZE 8388608 CACHE STRING "Stack size (default is 8MB)")
 
 if(PROFILE)
   SET(GCC_COVERAGE_COMPILE_FLAGS "-g -O0 --coverage -fprofile-arcs -ftest-coverage -fprofile-dir=.gcov")
   SET(GCC_COVERAGE_LINK_FLAGS    "--coverage")
+  # TODO use more specific commands "target_compile_options" etc. to avoid potentially affecting submodules
   SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS}")
   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS}")
   SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${GCC_COVERAGE_LINK_FLAGS}")
   # Make sure to override default optimization for Release type
   string(REPLACE "-O3" "-O0" CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
 else()
   # Use same optimization as qmake build for Release
-  string(REPLACE "-O3" "-O2" CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
+  #string(REPLACE "-O3" "-O2" CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
 endif()
 
 if (NULLGL)
   set(HEADLESS ON)
 endif()
 if(EXPERIMENTAL)
-  add_definitions(-DENABLE_EXPERIMENTAL)
+  target_compile_definitions(OpenSCAD PRIVATE ENABLE_EXPERIMENTAL)
 endif()
 if(IDPREFIX)
-  add_definitions(-DIDPREFIX)
+  target_compile_definitions(OpenSCAD PRIVATE IDPREFIX)
 endif()
 
 if (NOT "${SUFFIX}" STREQUAL "")
   set(SUFFIX_WITH_DASH "-${SUFFIX}")
-  add_definitions(-DOPENSCAD_SUFFIX="${SUFFIX_WITH_DASH}")
+  target_compile_definitions(OpenSCAD PRIVATE OPENSCAD_SUFFIX="${SUFFIX_WITH_DASH}")
 endif()
 
 if (MXECROSS)
-  add_definitions(-DGLEW_STATIC)
-  add_definitions(-DLIBXML_STATIC)
-  add_definitions(-DSTATIC_QT_SVG_PLUGIN)
+  target_compile_definitions(OpenSCAD PRIVATE GLEW_STATIC LIBXML_STATIC STATIC_QT_SVG_PLUGIN)
   list(APPEND PLATFORM_LIBS -lbcrypt)
 endif()
 
 if (MXECROSS OR WIN32)
   add_subdirectory(winconsole)
-  set(CMAKE_WIN32_EXECUTABLE ON)
+  set_property(TARGET OpenSCAD PROPERTY WIN32_EXECUTABLE ON)
 endif()
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/objects")
-make_directory(${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+set(OPENSCAD_LIB_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/objects")
+set_property(TARGET OpenSCAD PROPERTY LIBRARY_OUTPUT_DIRECTORY ${OPENSCAD_LIB_OUTPUT_DIR})
```

Comment:
I guess not, I was just keeping it so there would be minimal modifications to behavior, without really thinking too hard about it.

It looks like the only things that end up in that directory are the lexer/parser generated .cxx and ,hxx files. but those are explicitly given the path in their bison/flex calls.

I mean, it has a specific [meaning in cmake](https://cmake.org/cmake/help/latest/prop_tgt/LIBRARY_OUTPUT_DIRECTORY.html), but I guess it doesn't matter since we have no library targets (i.e. no `add_library` commands in our `CMakeListst.txt`) 
I suppose that mimalloc is now be the only library that we build along with openscad, but it's output is going into `[build]/submodules/mimalloc/` and not this `objects` path, which still seems fine to me.  

Would you rather just have it removed?

Reply:
Removing what's not strictly needed is probably a good idea, the setup is complicated enough so without `qmake` to consider anymore we can freely move into more `cmake` defaults.

Reply:
I read the docs and agree that this change is effectively a no-op from the existing config, so it's probably safer to leave it as you've suggested.